src/troute-config/troute/config/output_parameters.py

```diff
@@ -93,11 +93,19 @@ class ParityCheckCompareFileSet(BaseModel):
 
 class StreamOutput(BaseModel):
     # NOTE: required if writing StreamOutput files
-    stream_output_directory: Optional[DirectoryPath] = None
+    stream_output_directory: Optional[str] = None
     mask_output: Optional[FilePath] = None
     stream_output_time: int = 1
     stream_output_type:streamOutput_allowedTypes = ".nc"
     stream_output_internal_frequency: Annotated[int, Field(strict=True, ge=5)] = 5
+    
+    @validator('stream_output_directory', pre=True, always=True)
```

Comment:
A few minor improvements:

- We only need to validate after `pydantic` has coerced the input into a `pathlib.Path` (drop `pre=True` and `always=True`)
- Handle if we got an unexpanded path (e.g. `~/output/dir`)
- Raise if we got a file
- Create dir if it doesn't exist, and all intermediate directories

```python
    @validator('stream_output_directory')
    def validate_stream_output_directory(cls, value):
        if value is None:
            return None
            
        # expand ~/output/dir -> /home/user/output/dir
        value = value.expanduser()
        
        if value.exists() and not value.is_dir():
            raise ValueError(f"'stream_output_directory'={value!s} is a file, expected directory.")

        # make directory (and intermediates) if they don't exist
        value.mkdir(parents=True, exist_ok=True)
        return value
```